Autopilot/Inc/SDCard.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * SDcard fuctions
+ * Author: Avani Joshi and Laye Mehta
+ */
+
+#ifndef SDCARDHEADER_HPP
+#define SDCARDHEADER_HPP
+
+#include "SPI.hpp"
```

Comment:
SPI hal drivers should be included in the .cpp file instead of the header file. Reasoning is that if they're in the header file, anyone who includes your module suddenly knows about the hardware. Not only does that potentially mess up Simulation and unit test builds (which are strictly off the hardware) but it also means there is one more dependency, which is one more place where things can go wrong and one more thing someone using your module potentially has to worry about.

The general rule is to keep interfaces as simple and opaque as possible. That means users can very clearly understand how to use it and what it does for them without having to worry about the magic that goes on inside.  

I also sent ya'll a discord message about some issues about using the SPI.hpp header file rather than the ones directly generated by the hal.